pennylane/transforms/batch_input.py

```diff
@@ -13,18 +13,19 @@
 """
 Batch transformation for multiple (non-trainable) input examples following issue #2037
 """
-from typing import Union, Sequence, Callable, Tuple
+from typing import Callable, Sequence, Tuple, Union
 
 import pennylane as qml
 from pennylane import numpy as np
+from pennylane.tape import QuantumTape
 from pennylane.transforms.batch_transform import batch_transform
 
 
 @batch_transform
 def batch_input(
-    tape: Union[qml.tape.QuantumTape, qml.QNode],
-    argnum: Union[Sequence[int], int] = 0,
-) -> Tuple[Sequence[qml.tape.QuantumTape], Callable]:
+    tape: Union[QuantumTape, qml.QNode],
+    argnum: Union[Sequence[int], int],
```

Comment:
Conceptually a very nice call to remove the default! :)
However, this makes the PR a breaking change, I am not entirely sure we want that? :thinking:

Reply:
Why wouldn't we want a breaking change? I can keep this default value if needed, but it doesn't make much sense with the re-definition of the `argnum` argument.

Reply:
Maybe @antalszava or @josh146 can give their opinion?

Reply:
I am a bit flexible here --- if this is not a feature we expect many users to have been using, I am okay with a breaking change as long as it is clearly detailed in the changelog (in the breaking changes section), alongside examples of how to return to pre-existing behaviour with the new UI.

For more integral/highly used components of PL, it is best practice to go through a deprecation cycle for one release.

But in either case, we should definitely always _document_ the breaking changes in the changelog

Reply:
@dwierichs do you believe this is a widely used feature? If not I will implement the change and add it as a Breaking Change in the changelog.

Reply:
Same thoughts as what Josh mentioned. Overall, this is a user-facing feature and would like to give it a deprecation cycle ideally, but likely not too many users have started using it since v0.22.0 when it was first released.

Reply:
Breaking change has been added in the changelog.

Reply:
I can't tell whether it is widely used, but given that the fix is very easy and it hasn't been around for too long, I think we're good to go with this breaking change :)